src/Symfony/Bundle/FrameworkBundle/KernelBrowser.php

```diff
@@ -107,10 +107,7 @@ public function enableReboot()
         $this->reboot = true;
     }
 
-    /**
-     * @param UserInterface $user
-     */
-    public function loginUser($user, string $firewallContext = 'main'): self
+    public function loginUser(UserInterface $user, string $firewallContext = 'main'): self
```

Comment:
This should be reverted (see the first if statement)

```suggestion
    public function loginUser($user, string $firewallContext = 'main'): self
```